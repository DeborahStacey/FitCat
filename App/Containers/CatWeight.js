import React from 'react'
import { Text, View, TextInput } from 'react-native'
import I18n from 'react-native-i18n'
import { Colors } from '../Themes'
import { default as WellCatManager } from '../Services/WellCatManager'
import RoundedButton from '../Components/RoundedButton'
import styles from './Styles/CatWeightStyle'

export default class CatWeight extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      weight: 0,
      inputWeight: 0
    }
    this.getWeight()
  }

  async getWeight () {
    WellCatManager.getActiveCat().then((result) => {
      console.log(result)
      this.setState({
        weight: result.pet.weight
      })
    })
  }

  updateWeightPress = () => {
    WellCatManager.updateWeight(this.state.inputWeight)
    this.getWeight()
  }

  render () {
    return (
      <View style={styles.mainContainer}>
        <Text style={styles.weightText}>Current Weight: {this.state.weight}</Text>
        <View>
          <TextInput
            style={styles.updateInput}
            placeholder={I18n.t('update_weight_placeholder')}
            placeholderTextColor={Colors.placeholderText}
            autoCapitalize={'none'}
            onChangeText={(inputWeight) => this.setState({inputWeight})}
            keyboardType='numeric'
          />
        </View>
        <View style={styles.updateButtonView}>
          <RoundedButton onPress={() => this.updateWeightPress()}>
            {I18n.t('update_weight')}
          </RoundedButton>
        </View>
      </View>
    )
  }
}
