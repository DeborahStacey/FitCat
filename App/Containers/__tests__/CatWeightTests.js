import React from 'react'
import CatWeight from '../CatWeight'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders the cat weight page correctly', () => {
  const tree = renderer.create(
    <CatWeight />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
