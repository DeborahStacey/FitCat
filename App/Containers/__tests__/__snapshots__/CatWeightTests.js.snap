exports[`test renders the cat weight page correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F5FF",
      "flex": 1,
      "marginTop": 64,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "paddingTop": 20,
        "textAlign": "center",
      }
    }>
    [missing \"en.current_weight\" translation]
     
    0
     lbs
  </Text>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginHorizontal": 25,
        }
      }>
      <View
        style={
          Object {
            "flex": 1,
          }
        }>
        <TextInput
          autoCapitalize="none"
          keyboardType="numeric"
          maxLength={6}
          onChangeText={[Function]}
          placeholder="[missing \"en.update_weight_placeholder\" translation]"
          placeholderTextColor="#787A7F"
          style={
            Object {
              "alignSelf": "stretch",
              "borderColor": "gray",
              "borderWidth": 1,
              "height": 40,
              "marginVertical": 10,
              "minWidth": 120,
              "paddingLeft": 10,
            }
          } />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#466ECC",
              "borderRadius": 5,
              "height": 45,
              "justifyContent": "center",
              "marginHorizontal": 25,
              "marginVertical": 10,
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 14,
                "fontWeight": "bold",
                "marginHorizontal": 5,
                "marginVertical": 10,
                "textAlign": "center",
              }
            }>
            [MISSING \"EN.UPDATE_WEIGHT\" TRANSLATION]
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#D8D8D8",
        "borderBottomWidth": 1,
        "borderTopColor": "#D8D8D8",
        "borderTopWidth": 0.5,
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "paddingTop": 20,
        "textAlign": "center",
      }
    }>
    [missing \"en.weight_history\" translation]
  </Text>
  <ScrollView />
</View>
`;
